cocotb/result.py

```diff
@@ -83,7 +83,7 @@ def __init__(self, *args, **kwargs):
         self.stderr = StringIO()
 
 
-class ExternalException(StopIteration):
+class ExternalException(Exception):
```

Comment:
Earlier in your series you had a patch with "remove  ExternalException". I didn't look through the whole tree, but is it removed everywhere and just kept for backward-compat here, can  ExternalException be removed here as well, or is there anything I'm missing?

Reply:
Just kept for backwards-compatibility at this point.

Reply:
To be clear, that path should have said "Remove uses of ExternalException"